source/qb64pe.bas

```diff
@@ -24607,6 +24636,39 @@ SUB Set_OrderOfOperations
     i = i + 1: OName(i) = ",": PL(i) = 1000
 
     REDIM _PRESERVE OName(i) AS STRING, PL(i) AS INTEGER
+
+
+
+    'init the suffix conversions once here and be done with them as well.
+    'and the below is a conversion list so symbols don't get cross confused.
+    REDIM _PRESERVE PP_TypeMod(27) AS STRING, PP_ConvertedMod(27) AS STRING
+    PP_TypeMod(1) = "~`": PP_ConvertedMod(1) = "C_UBI" 'unsigned bit
+    PP_TypeMod(2) = "~%%": PP_ConvertedMod(2) = "C_UBY" 'unsigned byte
+    PP_TypeMod(3) = "~%&": PP_ConvertedMod(3) = "C_UOF" 'unsigned offset
+    PP_TypeMod(4) = "~%": PP_ConvertedMod(4) = "C_UIN" 'unsigned integer
+    PP_TypeMod(5) = "~&&": PP_ConvertedMod(5) = "C_UIF" 'unsigned integer64
+    PP_TypeMod(6) = "~&": PP_ConvertedMod(6) = "C_ULO" 'unsigned long
+    PP_TypeMod(7) = "`": PP_ConvertedMod(7) = "C_BI" 'bit
+    PP_TypeMod(8) = "%%": PP_ConvertedMod(8) = "C_BY" 'byte
+    PP_TypeMod(9) = "%&": PP_ConvertedMod(9) = "C_OF" 'offset
+    PP_TypeMod(10) = "%": PP_ConvertedMod(10) = "C_IN" 'integer
+    PP_TypeMod(11) = "&&": PP_ConvertedMod(11) = "C_IF" 'integer64
+    PP_TypeMod(12) = "&": PP_ConvertedMod(12) = "C_LO" 'long
+    PP_TypeMod(13) = "!": PP_ConvertedMod(13) = "C_SI" 'single
+    PP_TypeMod(14) = "##": PP_ConvertedMod(14) = "C_FL" 'float
+    PP_TypeMod(15) = "#": PP_ConvertedMod(15) = "C_DO" 'double
+    PP_TypeMod(16) = "_RGB32": PP_ConvertedMod(16) = "C_RG" 'rgb32
+    PP_TypeMod(17) = "_RGBA32": PP_ConvertedMod(17) = "C_RA" 'rgba32
+    PP_TypeMod(18) = "_RED32": PP_ConvertedMod(18) = "C_RX" 'red32
+    PP_TypeMod(19) = "_GREEN32": PP_ConvertedMod(19) = "C_GR" 'green32
+    PP_TypeMod(20) = "_BLUE32": PP_ConvertedMod(20) = "C_BL" 'blue32
+    PP_TypeMod(21) = "_ALPHA32": PP_ConvertedMod(21) = "C_AL" 'alpha32
+    PP_TypeMod(22) = "RGB32": PP_ConvertedMod(22) = "C_RG" 'rgb32
```

Comment:
```
    FOR j = 1 TO UBOUND(PP_TypeMod)
        IF qb64prefix_set = 0 AND j > 21 THEN EXIT FOR 'only check the no_prefix junk if necessary
```

We only check those extra noprefix values if the flag is set.  Sure, the array might always hold those values, but we just exit the check once we get to them.

This seems more efficient and logical to me than the way it was storing two variables and then choosing which variable to use, which just obfuscates the code in that area making it more complex than it needs to be.

Reply:
Oh I see - why are we hardcoding the `21` rather than place it in a variable like it was before?

Reply:
I can always make this a shared variable between those two subs, if you prefer, but it just makes more sense to me to have 1 variable for the cutoff point here, rather than 2 variables and then an if check to toggle between them as it was.  Heck even the following would make more sense and be easier to sort out: 

IF qb64prefext_set THEN j_limit = UBOUND(PP_TypeMod) ELSE j_limit = 21 'or a shared variable at that cutoff point
FOR j = 1 to j_limit

Reply:
Yeah that part is fine, I just think the `21` shouldn't be hardcoded, we should set a variable for it back where the array is initialized.